src/copy.jl

```diff
@@ -80,7 +85,8 @@ function copy_model(model::Model)
     # changed, the `index_map` gives the map between the indices of
     # `backend(model` and the indices of `backend(new_model)`.
     index_map = MOI.copy_to(backend(new_model), backend(model),
-                            copy_names = true)
+                            copy_names = true, 
+                            filter_constraints = filter_constraints)
```

Comment:
There's a problem here. `MOI.copy_to` expects the filter to be on `MOI.ConstraintIndex` objects, while a JuMP user is likely to write a filter based on `JuMP.ConstraintIndex` (e.g., see the tests written). Some translation needs to be done.